File: front/src/app/shared/model/filter/evento-filter.ts
import {BaseFilter} from '@/shared/model/filter/base-filter';

export class EventoFilter extends BaseFilter {

    nome?: string;
    clienteId?: number;
    status?: string;

    constructor(init?: Partial<EventoFilter>) {
        super();
        Object.assign(this, init);
    }
}

File: back/src/main/java/br/eng/eliseu/presente/model/filter/EventoFilter.java
package br.eng.eliseu.presente.model.filter;

import br.eng.eliseu.presente.model.StatusEnum;
import br.eng.eliseu.presente.model.core.BaseFilter;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class EventoFilter extends BaseFilter {

    private Long id;
    private String nome;
    private Long clienteId;
    private StatusEnum status;

}
